### PR TITLE
Create function to compute BFT headers on block creation - Closes #4233

### LINKED ISSUE #4233: Create function to compute BFT headers on block creation
### Description
When forging a new block, we need to compute two new properties and include them in the header:
- `maxHeightPreviouslyForged`
- `prevotedConfirmedUptoHeight` 

Currently, the `BFT` module doesn't provide an interface to compute those properties so **this must be implemented**

### Motivation
We need those properties in order to create a block successfully under the new protocol.

### Acceptance Criteria
- These two properties are properly computed and edge cases are considered.
- Unit tests are implemented

### Additional Information

### PR DESCRIPTION
**Pull request by limiaspasdaniel**
### What was the problem?
There was no way to compute these BFT headers required for block forging.
### How did I solve it?
By implementing a function to handle this computation inside `BFT` module.
### How to manually test it?
Run Jest tests on `bft.spec.js` file.
### Review checklist

- [ ] The PR resolves #4233 
- [ ] All new code is covered with unit tests
- [ ] Relevant integration / functional tests are added
- [ ] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [ ] Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/bft/bft.js [hunk 1]
```diff
@@ -141,6 +142,28 @@ class BFT extends EventEmitter {
...
 	}
 
+	/**
+	 * Computes maxHeightPreviouslyForged and prevotedConfirmedUptoHeight properties that are necessary
+	 * for creating a new block
+	 * @param delegatePubKey
+	 * @return {Promise<{prevotedConfirmedUptoHeight: number, maxHeightPreviouslyForged: (number|*)}>}
+	 */
+	async computeHeadersForNewBlock(delegatePubKey) {
+		const blockHeaders = cloneDeep(this.finalityManager.headers.items);
```
Comment (SargeKhan) [inline framework/src/modules/chain/bft/bft.js]: Why are we required to clone the headers?

File: framework/src/modules/chain/bft/bft.js [hunk 2]
```diff
@@ -141,6 +142,28 @@ class BFT extends EventEmitter {
...
+		const lastBlockForgedByDelegate = blockHeaders
+			.reverse()
+			.find(blockHeader => blockHeader.delegatePublicKey === delegatePubKey);
```
Comment (SargeKhan) [inline framework/src/modules/chain/bft/bft.js]: Is this method of getting the lastBlockForgedByDelegate valid in all cases? Imagine a scenario where a delegate forged on a forked chain. In this case, the `lastBlockForgedByDelegate` should correspond to that block; this block will not be part of the blockHeaders.

File: framework/test/jest/unit/specs/modules/chain/bft/bft.spec.js [hunk 3]
```diff
@@ -386,6 +391,134 @@ describe('bft', () => {
...
 		});
 
+		describe('computeHeadersForNewBlock()', () => {
+			let bft;
+
+			beforeEach(async () => {
+				bft = new BFT(bftParams);
+
+				jest
+					.spyOn(bft, '_getLastBlockHeight')
```
Comment (SargeKhan) [inline framework/test/jest/unit/specs/modules/chain/bft/bft.spec.js]: Do we need to spy on mocked implementations?
